#! /bin/sh -e

# All lines beginning with `# DPATCH:' are a description of the patch.
# DP: fix prototype clash with linux-kernel-headers version 2.6.18 in
# DP: psignal definition

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.

--- gcc-4.1.1/libiberty/strsignal.c.orig	2007-02-06 16:39:02.000000000 +0300
+++ gcc-4.1.1/libiberty/strsignal.c	2007-02-06 16:39:26.000000000 +0300
@@ -549,7 +549,7 @@
 #ifndef HAVE_PSIGNAL
 
 void
-psignal (unsigned signo, char *message)
+psignal (int signo, const char *message)
 {
   if (signal_names == NULL)
     {
--- src/gcc/fix-header.c.orig	2007-04-04 14:17:15.000000000 +0400
+++ src/gcc/fix-header.c	2007-04-04 14:18:32.000000000 +0400
@@ -666,6 +666,7 @@
     exit (FATAL_EXIT_CODE);
 
   register_include_chains (scan_in, NULL /* sysroot */, NULL /* iprefix */,
+		  	   NULL /* imultilib */,
 			   true /* stdinc */, false /* cxx_stdinc */,
 			   false /* verbose */);
 
